lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -2456,21 +2459,39 @@ Value::ValueType SwiftLanguageRuntimeImpl::GetValueType(
         return Value::ValueType::LoadAddress;
       }
 
-      if (auto *ts = llvm::dyn_cast_or_null<TypeSystemSwift>(
-              dynamic_type.GetTypeSystem()))
-        switch (ts->GetAllocationStrategy(dynamic_type.GetOpaqueQualType())) {
-        case SwiftASTContext::TypeAllocationStrategy::eDynamic:
-        case SwiftASTContext::TypeAllocationStrategy::eUnknown:
-          break;
-        case SwiftASTContext::TypeAllocationStrategy::eInline: // inline data;
-                                                               // same as the
-                                                               // static data
+      lldb::addr_t existential_address;
+      bool use_local_buffer = false;
+
+      if (in_value.GetValueType() == eValueTypeConstResult &&
+          static_value_type == lldb_private::Value::ValueType::HostAddress) {
+        existential_address = in_value.GetValue().GetScalar().ULongLong();
+        use_local_buffer = true;
+      } else {
+        existential_address = in_value.GetAddressOf();
+      }
+
+      if (use_local_buffer)
+        PushLocalBuffer(existential_address,
+                        in_value.GetByteSize().getValueOr(0));
+
+      swift::remote::RemoteAddress remote_existential(existential_address);
+      auto *reflection_ctx = GetReflectionContext();
+      llvm::Optional<bool> maybe_is_inlined =
+          reflection_ctx->isValueInlinedInExistentialContainer(
+              remote_existential);
+
+      if (use_local_buffer)
+        PopLocalBuffer();
+
+      if (maybe_is_inlined) {
```

Comment:
It's a bit confusing because maybe_is_inlined is both a bool and an optional and both the value and wrapper behave like a bool, and the "maybe" sounds like it's part of the value rather than the wrapper. How about:

if (!is_inlined.HasValue()) {
   // An error occurred, default to address.
   return Value::ValueType::LoadAddress;
}

if (*is_inlined) {
    // Inlined data, same as static data.
    return static_value_type;
}

HasValue makes it more obvious that we are checking the optional container. What do you think?